### ISSUE TITLE
How to delete old collections by time period?

### ISSUE THREAD

**Issue by ohsp123-arango** [arangodb/arangodb#16348]
## My Environment

* __ArangoDB Version__:        3.7.2 Enterprise
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:    Manual Start
* __Configuration__:               three nodes cluster
* __Infrastructure__:               own serevrs
* __Operating System__:        REHL 8.2
* __Total RAM in your machine__:        32GB each cluster nodes
* __Disks in use__:        400GB for each cluster nodes
* __Used Package__:              RedHat .rpm

## Component, Query & Data

__Affected feature__:

__AQL query (if applicable)__:
```


```
__AQL explain and/or profile (if applicable)__:

__Dataset__:

__Size of your Dataset on disk__:
400GB 

__Replication Factor & Number of Shards (Cluster only)__:
1

## Steps to reproduce

1. 
2. 
3. 

__Problem__: It's not easy to delete obsolete collections because the collection names are encrypted and not easy to recognize.

__Expected result__: Can anyone advise me on how can I delete collections by the period of time? (e.g using arangosh)

**Comment by dothebart** [comment]
From the top of my head could be looking somewhat like this:
```
db._collections().filter(name => { return name==='match' ; }.forEach(col => {db[col].delete();})
```

you could use `--javascript.execute-string` or a shebang script to run the above.

For reference:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

You can try that interactively in arangosh first.

**Comment by Simran-B** [comment]
What do you mean by "period of time"? Collection creation date?

**Comment by ohsp123-arango** [comment]
Yes, I mean how I can sort the collections by creation date and delete the old (e.g., more than six months aged) collections obsoleted.

**Comment by Simran-B** [comment]
Collections do not carry such metadata. See the related feature request #6048
As a hacky workaround, you could decode the document revisions, find the oldest timestamp for every collection, and base your decision on that - but it's obviously different to the collection creation date:

```js
FOR doc IN @@coll
  COLLECT AGGREGATE min = MIN(DECODE_REV(doc._rev).date)
  RETURN min
```
